### PR TITLE
Update readme to describe deployment options

### LINKED ISSUE #25: Roadmap in relation to the Helm Charts
Firstoff:
Design looks awesome, personally hate the overcomplicated ever-breaking s6 overlay.

But: What's the plan?
Is this already natively copy-pasta replacement for the images used in the Helm charts?
Is the goal to evenually be a replacement for us with the helmcharts (aka fully replace LinuxServer containers)?
If is, whats the timeframe we should reasonably expect for it to hit the helmcharts?

Is this a valid issue?
Yes.
TLDR:
Maybe a little readme update is in order? ;-)

### PR DESCRIPTION
**Pull request by Ornias1993**
As discussed in #25
Gives a little more insight about them being compatible with the k8s-at-home helmcharts.
(but not being limited to the helmcharts)

Fixes #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,14 @@ _There's so many images out there, why do I want to use these?_
 
 ## Deployment Information
 
+### Deployment on Kubernetes
+
+Most of these containers are frequently used as drop-in-replacements for our k8s-at-home Helm Charts.
+While we do not know if (or when) these containers will be the default for the Helm Charts, they work just fine!
+
+Within docker-compose and other solutions, they should work just fine too.
```
Comment (bjw-s) [inline README.md:24]:
```suggestion
Within docker-compose and other solutions, they should work just fine too but that isn't guaranteed. 
```

We don't want to give the impression of supporting other platforms 😉

File: README.md [hunk 2]
```diff
@@ -16,6 +16,14 @@ _There's so many images out there, why do I want to use these?_
 
 ## Deployment Information
 
+### Deployment on Kubernetes
+
+Most of these containers are frequently used as drop-in-replacements for our k8s-at-home Helm Charts.
```
Comment (bjw-s) [inline README.md:21]:
```suggestion
Most of these containers can be used as drop-in-replacements for the images in our k8s-at-home [Helm charts](https://github.com/k8s-at-home/charts/).
```

**Comment by onedr0p** [comment]
I am interested in if we should mention about `securityContext` since we do not implement s6-overlay

**Comment by Ornias1993** [comment]
> I am interested in if we should mention about `podSecurityContext` since we do not implement s6-overlay, it's not _really_ a drop in replacement because of permissions.
> 
> Switching to these images requires you to change permissions in the PVC [1] (`chown -R 568:568 /path/to/volume/`) or [2] setting the `podSecurityContext` in the helm charts to your user/group ids
> 
> [1] https://github.com/k8s-at-home/container-images/blob/main/ubuntu/Dockerfile#L16
> [2] =>
> 
> ```
> podSecurityContext:
>   runAsUser: 1001
>   runAsGroup: 1001
>   fsGroup: 1001
> ```

It might be prudent, as there is some confusion about it and we are actually promoting it here as a 'somewhat' drop-in replacement...

**Comment by onedr0p** [comment]
This is really the major difference in the images since k8s offers a way to handle permissions differently.

**Comment by Ornias1993** [comment]
> This is really the major difference in the images since k8s offers a way to handle permissions differently.

Yeah, one of my major grievences with S6 is the fact that in the meantime both docker-compose AND k8s handles it differently and they keep using their own permissionsystem....

I'll try and make a mockup for it :)

**Comment by onedr0p** [comment]
Awesome, it's maybe worth noting that from a fresh install using the helm chart and these images, people _shouldn't_ have to change perms with `chown` or set `podSecurityContext`. But I am not 100% with that being true with all CSI drivers.

**Comment by Ornias1993** [comment]
> Awesome, also it's maybe worth knowing that from a fresh install using the helm chart and these images, people _shouldn't_ have to change perms with `chown` or set `podSecurityContext`. But I am not 100% with that being true with all CSI drivers. rook-ceph had no issues where longhorn needed `podSecurityContext` set.
> 
> So maybe we can leave this bit out. Just more of a FYI

Thanks for the note, there is indeed a big difference between:
"Need to"
and
"Might Need to"

**Comment by onedr0p** [comment]
hahah for sure. It's hard to tell with all teh CSI drivers out there, we will know more over time.

@Ornias1993 join our Discord if you want :)

https://discord.gg/sTMX7Vh

**Comment by Ornias1993** [comment]
> hahah for sure. It's hard to tell with all teh CSI drivers out there, we will know more over time.

yeah thats an ongoing issue, though k8s is originally meant for people that know what they are doing... it doesn't make it easier to make this accessable to the home-crowd.

> @Ornias1993 join our Discord if you want :)
> 
> https://discord.gg/sTMX7Vh

Maybe i'll join oneday, i'm not much on Discord though...

**Comment by Ornias1993** [comment]
@onedr0p I've found one edgecase where chown is required that might be relevant to our discussion here:
hostPath.

While normally (with pvc's and pv's) the `fsgroup` automatically also chowns the mounted folder (as you described: chown shouldn't be needed), this is not the case when people mount storage using hostPath. This might be relevant to keep in mind and/or document.

**Pull request merged**